assets/model_monitoring/components/src/shared_utilities/store_url.py

```diff
@@ -77,39 +76,60 @@ def _get_url(self, scheme=None, store_type=None, relative_path=None) -> str:
+            from azure.ai.ml.identity import AzureMLOnBehalfOfCredential
+            print("Using AML OBO credential from StoreUrl.get_credential() because the internal datastore is None.")
             return AzureMLOnBehalfOfCredential()
         elif self._datastore.datastore_type == "AzureBlob":
             if self._datastore.credential_type == "AccountKey":
+                print("Using acount key credential from StoreUrl.get_credential() with blob datastore.")
                 return self._datastore.account_key
             elif self._datastore.credential_type == "Sas":
+                print("Using SAS token credential from StoreUrl.get_credential() with blob datastore.")
                 return AzureSasCredential(self._datastore.sas_token)
             elif self._datastore.credential_type is None or self._datastore.credential_type == "None":
+                from azure.ai.ml.identity import AzureMLOnBehalfOfCredential
+                print("Using AML OBO credential from StoreUrl.get_credential() with blob datastore"
+                      " where the saved credential type is null.")
                 return AzureMLOnBehalfOfCredential()
             else:
                 raise InvalidInputError(f"Unsupported credential type: {self._datastore.credential_type}, "
                                         "only AccountKey and Sas are supported.")
         elif self._datastore.datastore_type == "AzureDataLakeGen2":
             if self._datastore.tenant_id and self._datastore.client_id and self._datastore.client_secret:
+                print("Using Client Secret credential from StoreUrl.get_credential() with Gen2 datastore.")
                 return ClientSecretCredential(tenant_id=self._datastore.tenant_id, client_id=self._datastore.client_id,
                                               client_secret=self._datastore.client_secret)
             else:
+                from azure.ai.ml.identity import AzureMLOnBehalfOfCredential
+                print("Using AML OBO credential from StoreUrl.get_credential() with Gen2 datastore"
+                      " where the saved credential info does not have client secret available to authenticate with.")
                 return AzureMLOnBehalfOfCredential()
         else:
             raise InvalidInputError(f"Unsupported datastore type: {self._datastore.datastore_type}, "
                                     "only Azure Blob and Azure Data Lake Gen2 are supported.")
 
     def is_credentials_less(self) -> bool:
         """Check if the store url is credential less."""
-        credential = self.get_credential()
+        # TODO: remove after we figure out cache failure issues.
+        # Should be able to import AzureMLOnBehalfOfCredential to check the class w/o issues.
+        credential = None
+        try:
+            credential = self.get_credential()
+            from azure.ai.ml.identity import AzureMLOnBehalfOfCredential
+        except ModuleNotFoundError:
```

Comment:
suggest to arrange the try except as below:
```py
try:
  credential = self.get_credential()
  from ... import AmlOboCrednetial
  return credetnial is Non or isinstance(...)
except ModuleNotFoundError:
  print(...)
  return credential is None
```

Reply:
Done.